lib/caasp.pm

```diff
@@ -101,6 +102,34 @@ sub trup_call {
     wait_serial 'trup-1-' if $check == 2;
 }
 
+# Reboot if there's a diff between the current FS and the new snapshot
+sub check_reboot_changes {
+    my $change_expected = shift // 1;
+
+    # Compare currently mounted and default subvolume
+    my $time    = time;
+    my $mounted = "mnt-$time";
+    my $default = "def-$time";
+    assert_script_run "mount | grep 'on / ' | egrep -o 'subvolid=[0-9]*' | cut -d'=' -f2 > $mounted";
+    assert_script_run "btrfs su get-default / | cut -d' ' -f2 > $default";
+    my $change_happened = script_run "diff $mounted $default";
+
+    # If changes are expected check that default subvolume changed
+    die "Error during diff" if $change_happened > 1;
+    die "Change expected: $change_expected, happeed: $change_happened" if $change_expected != $change_happened;
+
+    # Reboot into new snapshot
+    process_reboot 1 if $change_happened;
+}
+
+# Install a pkg in Kubic
```

Comment:
rebootmgr has to be turned off as prerequisity for this to work - at least put it in comment. Also you can use "process_reboot 1" instead of "check_reboot_changes" - it's faster and cleaner.

I am not sure if we need it in separate function when it's used only once - but that's up to you ;-)

Reply:
I've made the changes ;)